Source/bluetooth/HCISocket.h

```diff
@@ -598,7 +598,7 @@ namespace Bluetooth {
 
     public:
 
-        template<const uint16_t OPCODE, typename OUTBOUND, typename INBOUND, const uint8_t RESPONSECODE = ~0>
+        template<const uint16_t OPCODE, typename OUTBOUND, typename INBOUND, const int RESPONSECODE = ~0>
```

Comment:
I would like to see the error cause I think the receiving end should change. The HCI only will report an unsigned 8 bit value and now we are suggesting differently...

Reply:
This was the error:
`[build] [71/171  33% :: 8.477] Building CXX object Source/bluetooth/CMakeFiles/WPEFrameworkBluetooth.dir/HCISocket.cpp.o
[build] FAILED: Source/bluetooth/CMakeFiles/WPEFrameworkBluetooth.dir/HCISocket.cpp.o 
[build] /bin/g++-9  -DCORE_BLUETOOTH -DPLATFORM_PC_UNIX=1 -DWPEFrameworkBluetooth_EXPORTS -D_THUNDER_DEBUG -D__NO_WCHAR_SUPPORT__ -I../Source/bluetooth -I../Source/bluetooth/.. -I../Source/bluetooth/drivers -I../Source/core -I../Source/core/.. -I../Source/core/Source -I../Source/tracing -I../Source/tracing/.. -g -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=c++11 -Wno-psabi -Wall -std=gnu++11 -MD -MT Source/bluetooth/CMakeFiles/WPEFrameworkBluetooth.dir/HCISocket.cpp.o -MF Source/bluetooth/CMakeFiles/WPEFrameworkBluetooth.dir/HCISocket.cpp.o.d -o Source/bluetooth/CMakeFiles/WPEFrameworkBluetooth.dir/HCISocket.cpp.o -c ../Source/bluetooth/HCISocket.cpp
[build] In file included from ../Source/bluetooth/HCISocket.cpp:20:
[build] ../Source/bluetooth/HCISocket.h:819:114: error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned char’ [-Wnarrowing]
[build]   819 |             typedef CommandType<cmd_opcode_pack(OGF_LINK_CTL, OCF_CREATE_CONN), create_conn_cp, evt_conn_complete>
[build]       | `